server/src/events.mjs

```diff
@@ -87,70 +100,152 @@ function getRegisteredEventListener(userId, method) {
+    registration: {},
+    unRegistration: {},
+    method: methodName
+  };
+
+   // If isOn is true, add oMsg to metadata.registration
+   if (isOn) {
+    metadata.registration = oMsg;
+  } else {
+    // If isOn is false, add oMsg to metadata.unRegistration
+    metadata.unRegistration = oMsg;
+  }
+
+  return metadata;
 }
-// Is the given (incoming) message one that enables or disables an event listener?
-// Example: {"jsonrpc":"2.0","method":"lifecycle.onInactive","params":{"listen":true|false},"id":1}
-// Key: The 'on' in the unqualified method name, and (2) the params.listen parameter (regardless of true|false)
-function isEventListenerMessage(oMsg) {
-  const fqMethodName = oMsg.method
-  const methodName = fqMethodName.substring(fqMethodName.lastIndexOf('.') + 1);
-  return ( methodName.startsWith('on') && 'listen' in oMsg.params );
+
+
+/**
+ * Determines if the given object message is an event listener message.
+ * @param {object} oMsg - The object message to check.
+ * @param {object} config - The configuration object to use.
+ * @returns {boolean} - True if the message is an event listener message, false otherwise.
+*/
+function isEventListenerMessage(oMsg, config) {
+  // Call extractEventData and store the result
+  const eventData = extractEventData(oMsg, config);
+  
+  return (eventData !== false);
 }
 
-// Is the given (incoming) message one that enables an event listener?
-// Example: {"jsonrpc":"2.0","method":"lifecycle.onInactive","params":{"listen":true},"id":1}
-// Key: The params.listen === true value
+/**
+ * Determines whether the given message is intended to enable an event listener.
+ * A regex pattern and JSON path are provided in config to check whether the event listener is on.
+ * @param {Object} oMsg - The incoming message to check.
+ * @return {boolean} - Returns `true` if the message is intended to enable an event listener, `false` otherwise.
+ */
 function isEventListenerOnMessage(oMsg) {
-  if ( ! isEventListenerMessage(oMsg) ) { return false; }
-  return ( oMsg.params && oMsg.params.listen );
+  if (!isEventListenerMessage(oMsg, eventConfig.registrationMessage)) {
+    return false;
+  }
+  return true;
 }
 
-// Is the given (incoming) message one that disables an event listener?
-// Example: {"jsonrpc":"2.0","method":"lifecycle.onInactive","params":{"listen":false},"id":1}
-// Key: The params.listen === true value
+/**
+ * Determines whether the given message is intended to disable an event listener.
+ * A regex pattern and JSON path are provided in config to check whether the event listener is off.
+ * @param {Object} oMsg - The incoming message to check.
+ * @return {boolean} - Returns `true` if the message is intended to disable an event listener, `false` otherwise.
+ */
 function isEventListenerOffMessage(oMsg) {
-  if ( ! isEventListenerMessage(oMsg) ) { return false; }
-  return ( oMsg.params && ! oMsg.params.listen );
+  if (!isEventListenerMessage(oMsg, eventConfig.unRegistrationMessage)) {
```

Comment:
This seems overly complex.

I'd say we can make this a one-liner (return isEventListenerMessage(...)) but that also seems overly complex considering we can just use that one-liner where we'd want to call "isEventListenerOffMessage()".

Can this be simplified?

Reply:
Kept separate functions so code is easier to read/understand but simplified the functions into one liners